### DISCUSSION TITLE
seleniumbase (1.66.7) depends on requests (2.26.0)

### DISCUSSION THREAD

**Discussion by jonnyjimmy1** [seleniumbase/SeleniumBase#998] [Q&A] [answered]
Hello everyone, 

Is there a reason seleniumbase is pinned to requests 2.26?  What would be a workaround, if possible? 

 Because no versions of seleniumbase match >1.66.7,<2.0.0
   and seleniumbase (1.66.7) depends on requests (2.26.0), seleniumbase (>=1.66.7,<2.0.0) requires requests (2.26.0).
  And because prefect (0.15.6) depends on requests (>=2.20,<2.26)
   and no versions of prefect match >0.15.6,<0.16.0, seleniumbase (>=1.66.7,<2.0.0) is incompatible with prefect (>=0.15.6,<0.16.0).
  So, because allpdl-requirements depends on both prefect (^0.15.6) and seleniumbase (^1.66.7), version solving failed.

**Comment by mdmintz** [comment, marked as answer]
Hello @jonnyjimmy1, The latest version of ``requests`` is ``2.26.0`` for ``python_version>="3.6"``. SeleniumBase does what it can to always use the latest version of dependencies when available. Otherwise it would be very difficult to test all permutations/combinations of dependencies to see which combinations work and which ones don't. If the library you speak of isn't using the latest version of ``requests``, perhaps you can submit a request.

There are a few workarounds though. You can create a separate Python virtual environment for your SeleniumBase tests so that you aren't mixing dependencies. Or you can clone the repo and try your own combination of Python dependencies.

**Reply by jonnyjimmy1** [reply]
Hey Michael,  

Thanks for the quick response.  I really think you guys have done a great job with SeleniumBase, and I totally understand.  

 I'll get back to you on what we decide to do.
Thanks again,

**Reply by jonnyjimmy1** [reply]
Hey Michael, 

is there any way you could point me in the right direction on cloning the repo and try my own combination of Python dependencies.  Meaning, what do you think it would take to validate/verify/update seleniumbase to requests version for python_version>="3.7"?

Thanks
Jon

**Reply by mdmintz** [reply]
Hello @jonnyjimmy1 Cloning the repo and installing from that is simple (https://seleniumbase.io/help_docs/install/):
```bash
git clone https://github.com/seleniumbase/SeleniumBase.git
cd SeleniumBase/
pip install -e .
```
If you modify dependencies from the ``setup.py`` file, just rerun ``pip install -e .``.
Or if you install everything into a separate Python virtual environment, you might not need to worry about dependency conflicts.
Also, I just shipped version ``1.66.9`` minutes ago.
